compiler/il/OMRBlock.cpp

```diff
@@ -1472,11 +1472,35 @@ OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyE
             // If we can not find a register, store it into temp slot.
             if (!iter->second.second)
                {
-               iter->second.second = TR::Node::createWithSymRef(value, comp->il.opCodeForDirectLoad(value->getDataType()), 0, ref);
+               TR::DataType valDataType = value->getDataType();
+               TR::Node *convertedValueNode  = value;
+               // In case for a given node, if the load/store (Minimum 4 bytes for alignment) size in memory is not same as node's data type,
+               // We need to convert the node to the type which can be stored into memory. Also we need to make sure that replacement node is a
+               // node that is converted back to original type of the node.
+               if (comp->fe()->dataTypeForLoadOrStore(valDataType) != valDataType)
+                  {
+                  TR::ILOpCodes convertOpCode = TR::ILOpCode::getProperConversion(valDataType, comp->fe()->dataTypeForLoadOrStore(valDataType), false);
+                  convertedValueNode = TR::Node::create(convertOpCode, 1, value);
+                  }
+               TR::SymbolReference *storedValueSymRef = createSymRefForNode(comp, methodSymbol, convertedValueNode, self()->getExit());
+               self()->getExit()->insertBefore(TR::TreeTop::create(comp,
+                                                                     TR::Node::createWithSymRef(value,
+                                                                        comp->il.opCodeForDirectStore(convertedValueNode->getDataType()),
+                                                                        1,
+                                                                        convertedValueNode,
+                                                                        storedValueSymRef)));
+               TR::Node *replacementNode = TR::Node::createLoad(value, storedValueSymRef);
+               if (replacementNode->getDataType() != valDataType)
+                  {
+                  TR::ILOpCodes convertOpCode = TR::ILOpCode::getProperConversion(replacementNode->getDataType(), valDataType, false);
+                  replacementNode = TR::Node::create(convertOpCode, 1, replacementNode);
+                  }
+               iter->second.second = replacementNode;
                }
             }
          TR::Node *replacement = iter->second.second;
-         // Now we know where the value will be stored handle the two cases in step 3 based on the opcode type
+         // If the replacement is regLoad node, attach it to GlRegDep node of the entry node of the new block and
+         // exit node of the original block.
          if (replacement->getOpCode().isLoadReg())
```

Comment:
Given the reg case and the append used to be if-else is this still true? should the code be factored to reflect this to make it more clear?

Reply:
Looking into the code, I think you are correct, I can move this `if` block to the block where we have found the register candidate. It will be more clear.

Reply:
I have made the changes in https://github.com/eclipse/omr/pull/4551/commits/5712d5e16bfc4c66ac19c900086f36b788d1731c with updated comment.